test/libyul/yulOptimizerTests/functionGrouper/grouped_but_not_ordered.yul

```diff
@@ -0,0 +1,19 @@
+{
+    function f() -> y { y := 8 }
+    {
+        let x := 2
+    }
+}
+// ----
+// functionGrouper
+// {
+//     {
+//         {
```

Comment:
Depending on the policy wrt the flattener, it may make sense to try to avoid the additional block here, too... but I guess it's fine, if we apply the flatten anyways.

Reply:
I thought about inserting into the first statement if it is a block instead of creating a new block, but this way it is simpler.